wallet-api/src/Ledger/Value/TH.hs

```diff
@@ -42,166 +48,113 @@ data CurrencySymbol = CurrencySymbol Int
 
 makeLift ''CurrencySymbol
 
+curCmp :: Q (TExp (CurrencySymbol -> CurrencySymbol -> Ordering))
+curCmp = [|| \(CurrencySymbol i) (CurrencySymbol j) -> $$(P.intCompare) i j ||]
+
 currencySymbol :: Q (TExp (Int -> CurrencySymbol))
 currencySymbol = [|| CurrencySymbol ||]
 
 -- | Cryptocurrency value
 --   See note [Currencies]
-newtype Value = Value { getValue :: [(CurrencySymbol, Int)] }
+newtype Value = Value { getValue :: Map.Map CurrencySymbol Int }
     deriving (Show)
     deriving stock (Generic)
-    deriving anyclass (ToSchema, ToJSON, FromJSON)
+    deriving anyclass (ToJSON, FromJSON)
```

Comment:
The JSON instances should use the list-of-k-v-pairs representation. That way the consumers of contract endpoints (Playground and others) don't have to construct red-black trees as JSON objects.

An even better encoding would be as a JSON object, rather than a JSON list. This is how the the `ToJSON`/`FromJSON` instances for `Map k v` work (although that involves a typeclass `ToJSONKey` for converting `k` to/from a string).

Reply:
If you use the JSON encoding via (haha) `[(k, v)]` then that should also solve the `declareNamedSchema` issue.

Reply:
What's wrong with representing it with the derived instance?

Reply:
It exposes the details of the implementation.

```
*Wallet> Data.Aeson.encode (Ledger.Ada.adaValueOf 100)
"{\"getValue\":{\"tag\":\"Branch\",\"contents\":[\"B\",1,{\"tag\":\"Leaf\"},0,100,{\"tag\":\"Leaf\"}]}}"
```